src/_common/scroll/inview/parent.ts

```diff
@@ -1,43 +1,37 @@
 import Vue, { CreateElement } from 'vue';
 import { Component, Prop } from 'vue-property-decorator';
-import { ScrollWatcher } from '../watcher.service';
+import AppScrollScrollerTS from '../scroller/scroller';
 import { ScrollInviewContainer } from './container';
 
 @Component({})
 export class AppScrollInviewParent extends Vue {
-	@Prop(Number)
-	throttle?: number;
-
-	@Prop(Number)
-	velocity?: number;
-
-	@Prop(ScrollInviewContainer)
-	container?: ScrollInviewContainer;
-
-	// Don't have Vue watch these.
-	_scrollWatcher!: ScrollWatcher;
-	_container!: ScrollInviewContainer;
-
-	$el!: HTMLDivElement;
-
-	mounted() {
-		if (this.container) {
-			this._container = this.container;
-		} else {
-			this._scrollWatcher = new ScrollWatcher(this.$el);
-			this._container = new ScrollInviewContainer(
-				this._scrollWatcher,
-				this.throttle,
-				this.velocity
-			);
+	// If this is a child of AppScrollScroller, we need to get it as a prop so
+	// we can use that scroll element as the root context.
+	@Prop() scroller?: AppScrollScrollerTS;
```

Comment:
Nah, I tried it that way. Circular reference ruined that plan. Since the scroller is the only thing that creates the parents in the new code, I figured passing it through a ref would be good (note that this is pretty normal for frontend code as React would force you to do it this way anyway)

Reply:
Heh, figured it can't do this. oh well, this is fine, just a bit annoying.